capstone/src/test/java/com/google/sps/servlets/announcements/EditAnnouncementServletTest.java

```diff
@@ -0,0 +1,154 @@
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/** */
+@RunWith(JUnit4.class)
+public final class EditAnnouncementServletTest {
+
+  private final long SAMPLE_TIME = 123456789;
+  private final String SAMPLE_CLUB_NAME = "Test Club";
+  private final String SAMPLE_CONTENT = "A random announcement";
+  private final String SAMPLE_NEW_CONTENT = "A new announcement";
+  private final String TEST_EMAIL = "test-email@gmail.com";
+  private final String ID_PROP = "id";
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  private DatastoreService datastore;
+  private EditAnnouncementServlet servlet;
+
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig().setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+          new LocalUserServiceTestConfig(),
+          new LocalBlobstoreServiceTestConfig());
+
+  @Before
+  public void setUp() throws IOException {
+    helper.setUp();
+    MockitoAnnotations.initMocks(this);
+    this.servlet = new EditAnnouncementServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doPost_authorizedEdit() throws ServletException, IOException {
+    prepare();
+    helper.setEnvEmail(TEST_EMAIL).setEnvAuthDomain("google.com").setEnvIsLoggedIn(true);
+
+    String id = TEST_EMAIL + SAMPLE_CONTENT + SAMPLE_TIME;
+    when(request.getParameter(Constants.CONTENT_PROP)).thenReturn(SAMPLE_NEW_CONTENT);
+    when(request.getParameter(Constants.CLUB_PROP)).thenReturn(SAMPLE_CLUB_NAME);
+    when(request.getParameter(ID_PROP)).thenReturn(id);
+    servlet.doPost(request, response);
+
+    Query query = new Query(Constants.ANNOUNCEMENT_PROP);
+    PreparedQuery results = datastore.prepare(query);
+
+    String newId = TEST_EMAIL + SAMPLE_NEW_CONTENT + SAMPLE_TIME;
+    ImmutableList<Entity> entities =
+        Streams.stream(results.asIterable())
+            .filter(
+                entity ->
+                    SAMPLE_CLUB_NAME.equals(entity.getProperty(Constants.CLUB_PROP))
+                        && newId.equals(
+                            entity.getProperty(Constants.AUTHOR_PROP).toString()
+                                + entity.getProperty(Constants.CONTENT_PROP).toString()
+                                + entity.getProperty(Constants.TIME_PROP).toString()))
+            .collect(toImmutableList());
```

Comment:
Tests shouldn't be re-implementing the logic of the actual code. They should just be able to get the response and check the result directly. If that's a little hard to do because of the nature of the servlet structure, you can try using helper methods exposed to the test classes and specifically test those, like the doGetHelpers in other servlet classes.

Same below.